include/glibby/spatial/kdTree.h

```diff
@@ -0,0 +1,178 @@
+#pragma once
+
+#include <vector>
+#include <map>
+#include <queue>
+#include "glibby/primitives/point.h"
+#include <iostream>
+#include <string>
+
+namespace glibby
+{
+	template<typename T>
+	struct Node
+	{
+		T val;
+		Node<T>* left;
```

Comment:
use smart pointers instead! :)

Reply:
specifically, std::unique_ptr from the memory header
that way  you won't need the deletes and destructors anymore